### PR TITLE
Training code updates

### PR DESCRIPTION
**Pull request by guoweiqi**
Training code for enthalpy and RMSD training. Example config files and config options included

### CODE REVIEW

File: system/config_options.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+"run_type": "train" (training split) or "test" (testing split)
+"loss_type: "RMSD" (train to experimental RMSD) or "enthalpy (train to experimental enthalpy)
+"optimizer": "Adam" or "SGD" or "RMSProp" or "Adagrad" or "SM3" (which optimizer to use for gradient descent)
+"learning_rate": learning rate for gradient descent (1e-3 is a good place to start, but can be any float and heavily depends on the system/optimizer being used"
+"num_epochs": number of epochs in a training split, or number of tests in a testing split
+"batch_size": number of compounds per batch
+"dp_idxs": list of indices to specify which parameters to train to [0: bond k value, 1: ideal bond length, 2: angle k value, 3: ideal bond angle, 4: torsion k value, 5: torsion phase, 6: torsion period, 7: charges, 8: lj sigma, 9: lj epsilon]
+"guest_directory": path to directory containing the ligands for binding enthalpy or compounds for RMSD training
+"training_data": .npz (RMSD) or .csv (Enthalpy) file containing experimental data for training
```
Comment (proteneer) [inline system/config_options.txt]: What is the expected data layout of the .npz files?

File: system/forcefield.py [hunk 2]
```diff
@@ -18,16 +18,16 @@ def merge_potentials(nrgs):
...
-            c_nrgs.append(custom_ops.HarmonicBond_f32(a_args[0], a_args[1]))
-        elif a_name == custom_ops.HarmonicAngle_f32:
-            c_nrgs.append(custom_ops.HarmonicAngle_f32(a_args[0], a_args[1]))
-        elif a_name == custom_ops.PeriodicTorsion_f32:
-            c_nrgs.append(custom_ops.PeriodicTorsion_f32(a_args[0], a_args[1]))
-        elif a_name == custom_ops.LennardJones_f32:
-            c_nrgs.append(custom_ops.LennardJones_f32(a_args[0].astype(np.float32), a_args[1]))
-        elif a_name == custom_ops.Electrostatics_f32:
-            c_nrgs.append(custom_ops.Electrostatics_f32(a_args[0].astype(np.float32), a_args[1]))
+        if a_name == custom_ops.HarmonicBond_f64:
```
Comment (proteneer) [inline system/forcefield.py]: I take it the training was only stable under 64bit precision? Did we ever figure out why it didn't work in 32 bit?

File: system/training.py [hunk 3]
```diff
@@ -1,215 +1,1087 @@
...
+            '''.format(epoch,mean_loss,median_loss,time.time()-start_time))
+        
+    return losses, get_params(opt_state)
+        
+def rescale_and_center(conf, scale_factor=1):
+	'''
+	recenter ligands to the center of a-cd in host-guest set
+    '''
+    mol_com = np.sum(conf, axis=0)/conf.shape[0]
+    true_com = np.array([1.97698696, 1.90113478, 2.26042174]) # a-cd
```
Comment (proteneer) [inline system/training.py]: we may want to make this non-hardcoded

File: system/forcefield.py (hunk 2 above)
Comment (guoweiqi) [reply system/forcefield.py]: I recall the gradients blowing up in 32-bit precision, but I didn't look too deeply into the specific numerical error
